src/mock_miner.cpp

```diff
@@ -0,0 +1,201 @@
+    BlockIndex<Block>* tip = blockChain->getBestChain().tip();
+    assert(tip != nullptr);
+
+    Block block = miner->createNextBlock(*tip, {});
+    res = blockChain->acceptBlock(block, state);
+  }
+
+  return res;
+}
+
+VbkTx MockMiner::generateSignedVbkTx(const PublicationData& publicationData) {
+  VbkTx transaction;
+  transaction.signatureIndex = 7;
+  transaction.networkOrType.hasNetworkByte = true;
+  transaction.networkOrType.networkByte = 1;
+  transaction.networkOrType.typeId = (uint8_t)TxType::VBK_TX;
+  transaction.sourceAmount = Coin(1000);
+  transaction.sourceAddress = Address::fromPublicKey(defaultPublicKeyVbk);
+  transaction.publicKey = defaultPublicKeyVbk;
+  transaction.publicationData = publicationData;
+
+  auto hash = transaction.getHash();
+  transaction.signature =
+      veriBlockSign(hash, privateKeyFromVbk(defaultPrivateKeyVbk));
+
+  return transaction;
+}
+
+ATV MockMiner::generateValidATV(const PublicationData& publicationData,
+                                const VbkBlock::hash_t& lastKnownVbkBlockHash,
+                                ValidationState& state) {
+  ATV atv;
+  atv.transaction = generateSignedVbkTx(publicationData);
+  atv.merklePath.treeIndex = 1;
+  atv.merklePath.index = 0;
+  atv.merklePath.subject = atv.transaction.getHash();
+  atv.merklePath.layers = {atv.transaction.getHash()};
+
+  BlockIndex<vbk_block_t>* tip = vbk_blockchain->getBestChain().tip();
+
+  assert(tip != nullptr);
+
+  for (BlockIndex<VbkBlock>* walkBlock = tip;
+       walkBlock->header.getHash() != lastKnownVbkBlockHash;
+       walkBlock = walkBlock->pprev) {
+    atv.context.insert(atv.context.begin(), walkBlock->header);
+  }
+
+  atv.containingBlock =
+      vbk_miner->createNextBlock(*tip, atv.merklePath.calculateMerkleRoot());
+
+  vbk_blockchain->acceptBlock(atv.containingBlock, state);
+
+  return atv;
+}
+
+VbkPopTx MockMiner::generateSignedVbkPoptx(
+    const VbkBlock& publishedBlock,
+    const BtcBlock::hash_t& lastKnownBtcBlockHash,
+    ValidationState& state) {
+  VbkPopTx popTx;
+  popTx.networkOrType.hasNetworkByte = true;
+  popTx.networkOrType.networkByte = 1;
+  popTx.networkOrType.typeId = (uint8_t)TxType::VBK_POP_TX;
+  popTx.address = Address::fromPublicKey(defaultPublicKeyVbk);
+  popTx.publishedBlock = publishedBlock;
+
+  WriteStream stream;
+  popTx.publishedBlock.toRaw(stream);
+  popTx.address.getPopBytes(stream);
+
+  popTx.bitcoinTransaction = BtcTx(stream.data());
+  popTx.merklePath.index = 0;
+  popTx.merklePath.subject = popTx.bitcoinTransaction.getHash();
+  popTx.merklePath.layers = {popTx.bitcoinTransaction.getHash()};
+
+  BlockIndex<btc_block_t>* tip = btc_blockchain->getBestChain().tip();
+
+  assert(tip != nullptr);
+
+  for (BlockIndex<BtcBlock>* walkBlock = tip;
+       walkBlock->header.getHash() != lastKnownBtcBlockHash;
+       walkBlock = walkBlock->pprev) {
+    popTx.blockOfProofContext.insert(popTx.blockOfProofContext.begin(),
```

Comment:
the block order is reversed (the oldest block at blockOfProofContext[0]), should it be like this?

Reply:
No, they are should be in this order, from the lowest height of the block to the highest.